### PR TITLE
Bug 1820927 - Update Glean to v52.4.1

### PR DESCRIPTION
**Pull request by travis79**
This fixes a cargo audit vuln in the Glean dependency `remove_dir_all`

### Pull Request checklist ###

- [ ] **Quality**: This PR builds and tests run cleanly
  - Note:
    - For changes that need extra cross-platform testing, consider adding `[ci full]` to the PR title.
    - If this pull request includes a breaking change, consider [cutting a new release](https://github.com/mozilla/application-services/blob/main/docs/howtos/cut-a-new-release.md) after merging.
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes a changelog entry in [CHANGES_UNRELEASED.md](../CHANGES_UNRELEASED.md) or an explanation of why it does not need one
  - Any breaking changes to Swift or Kotlin binding APIs are noted explicitly
- [ ] **Dependencies**: This PR follows our [dependency management guidelines](https://github.com/mozilla/application-services/blob/main/docs/dependency-management.md)
  - Any new dependencies are accompanied by a summary of the due dilligence applied in selecting them.

[Branch builds](https://github.com/mozilla/application-services/blob/main/docs/howtos/branch-builds.md): add `[firefox-android: branch-name]` to the PR title.

### CODE REVIEW

**Review by tarikeshaq — CHANGES_REQUESTED** [review]
Thanks @travis79!! 

We should probably also bump the iOS dependency (by bumping the version in Xcode)

Also, if we'd like to fully fix the `remove_dir_all` advisory in application-services, we should run:
```sh
$ cargo update -p tempfile
```

`tempfile` has a dependency on the affected version, so updating that and landing the `Cargo.lock` changes should fix the CI here!

**Comment by travis79** [comment]
> Thanks @travis79!!
> 
> We should probably also bump the iOS dependency (by bumping the version in Xcode)
> 
> Also, if we'd like to fully fix the `remove_dir_all` advisory in application-services, we should run:
> 
> ```shell
> $ cargo update -p tempfile
> ```
> 
> `tempfile` has a dependency on the affected version, so updating that and landing the `Cargo.lock` changes should fix the CI here!

yeah yeah, I know....  😅   I found this out myself a little late. I'm getting a new release today.

**Comment by travis79** [comment]
> `During handling of the above exception, another exception occurred:`

**Comment by travis79** [comment]
@tarikeshaq this is ready for a re-review whenever you get a chance.

File: build.gradle
Comment (rvandermeulen) [inline build.gradle:11]: Should this be 52.4.1 now?

Comment (travis79) [reply build.gradle:11]: Yeah...  it should.

**Review by tarikeshaq — APPROVED** [review]
This LGTM! one thing we should do before landing is update the iOS dependency too in https://github.com/mozilla/application-services/blob/e27985157b0e664fcd14bd182eace1ce55720760/megazords/ios-rust/MozillaTestServices/MozillaTestServices.xcodeproj/project.pbxproj#L1029

(updating through Xcode is good enough)

I'm not blocking on it because that version is mostly only used for development and local tests 😄

**Comment by travis79** [comment]
> This LGTM! one thing we should do before landing is update the iOS dependency too in
> 
> https://github.com/mozilla/application-services/blob/e27985157b0e664fcd14bd182eace1ce55720760/megazords/ios-rust/MozillaTestServices/MozillaTestServices.xcodeproj/project.pbxproj#L1029
> 
> (updating through Xcode is good enough)
> 
> I'm not blocking on it because that version is mostly only used for development and local tests 😄

Might as well get it while I'm here, please stand by :)

**Pull request merged**
